src/qibo/backends/tensorflow.py

```diff
@@ -212,6 +212,18 @@ def zero_density_matrix(self, nqubits):
         state = self.tf.tensor_scatter_nd_update(state, idx, update)
         return state
 
+    def plus_state(self, nqubits):
+        state = self.tf.ones((2 ** nqubits,), dtype=self.dtype)
+        norm = self.tf.cast(self.np.sqrt(2 ** nqubits), dtype=self.dtype)
+        state = self.tf.math.scalar_mul(1/norm, state)
+        return state
+
+    def plus_density_matrix(self, nqubits):
+        state = self.tf.zeros(2 * (2 ** nqubits,), dtype=self.dtype)
+        norm = self.tf.cast(2 ** nqubits, dtype=self.dtype)
+        state = self.tf.math.scalar_mul(1/norm, state)
+        return state
```

Comment:
I have not tested this, but it may be possible to completely remove these methods if you use `self.np` in the respective methods in the numpy backend. This is because `self.np` becomes `tnp` in the tensorflow backend which is new feature in tensorflow and implements many numpy methods for `tf.Tensor` automatically.

Reply:
Yes, I agree. I've removed these methods. Thanks!